pkg/plugins/manager.go

```diff
@@ -0,0 +1,134 @@
+package pluginmanager
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/massalabs/thyra/pkg/config"
+)
+
+type PluginManifest struct {
+	Name    string
+	Bin     string
+	Version string
+}
+
+type PluginItem struct {
+	Manifest PluginManifest
+	Path     string
+	Port     int
+}
+
+type PluginManager struct {
+	HostPort    int
+	HostTLSPort int
+	plugins     []PluginItem
+}
+
+func (manager *PluginManager) List() []PluginItem {
+	return manager.plugins
+}
+
+func DetectPlugin(path string) (*PluginManifest, error) {
+	manifestFileName := "manifest.json"
+
+	manifestPath := filepath.Join(path, manifestFileName)
+
+	if _, err := os.Stat(manifestPath); err != nil {
+		errMsg := "unable to find manifest.json in " + path
+
+		return nil, errors.New(errMsg)
```

Comment:
this variable can be inlined, same for the 2 bellow.

```suggestion
		return nil, errors.New("unable to find manifest.json in " + path)
```